v2/config/env.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/golang/glog"
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	configMapName                = "kfp-launcher"
+	defaultPipelineRoot          = "minio://mlpipeline/v2/artifacts"
+	configKeyDefaultPipelineRoot = "defaultPipelineRoot"
+)
+
+// Config is the KFP runtime configuration.
+type Config struct {
+	data map[string]string
+}
+
+// FromConfigMap loads config from a kfp-launcher Kubernetes config map.
+func FromConfigMap(ctx context.Context, clientSet kubernetes.Interface, namespace string) (*Config, error) {
+	config, err := clientSet.CoreV1().ConfigMaps(namespace).Get(ctx, configMapName, metav1.GetOptions{})
+	if err != nil {
+		if k8errors.IsNotFound(err) {
+			glog.Infof("cannot find launcher configmap: name=%q namespace=%q", configMapName, namespace)
+			// LauncherConfig is optional, so ignore not found error.
+			return nil, nil
+		}
+		return nil, err
+	}
+	return &Config{data: config.Data}, nil
+}
+
+// Config.DefaultPipelineRoot gets the configured default pipeline root.
+func (c *Config) DefaultPipelineRoot() string {
+	// The key defaultPipelineRoot is optional in launcher config.
+	if c == nil || c.data[configKeyDefaultPipelineRoot] == "" {
+		return defaultPipelineRoot
+	}
+	return c.data[configKeyDefaultPipelineRoot]
+}
+
+// InPodNamespace gets current namespace from inside a Kubernetes Pod.
+func InPodNamespace() (string, error) {
+	// The path is available in Pods.
+	// https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#directly-accessing-the-rest-api
+	ns, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
What is the advantage of reading namespace from file instead of setting env variable in pod?
FYI:  https://github.com/kubernetes/kubernetes/pull/63707#issuecomment-539648137

Reply:
I'm not aware of what backwards compatibility requirement exists. We don't use "POD_NAMESPACE", so I don't think we need to be backward compatible with anything.